infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -106,7 +108,11 @@
     public LogUnitServer(ServerContext serverContext) {
         this.opts = serverContext.getServerConfig();
         this.serverContext = serverContext;
-        maxCacheSize = Utils.parseLong(opts.get("--max-cache"));
+        long maxCacheSizeFromUser = Utils.parseLong(opts.get("--max-cache"));
+
+        /* Ensure that maxCacheSize is at most CACHE_HEAP_RATIO * maxHeapSize */
```

Comment:
Right, but that's no different from supplying other wrong configs. I don't think we should mask bad config parameters. Also, if I was tweaking the server to figure out the best cache ratio, it doesn't make sense to recompile every time.